client/tests/integration/pagination.rs

```diff
@@ -13,14 +13,13 @@ fn limits_should_work() -> Result<()> {
 
     register_assets(&client)?;
 
-    let vec = &client
-        .build_query(asset::all_definitions())
+    let vec = client
+        .query(asset::all_definitions())
```

Comment:
I don't like it that on the http client we have API like `asset::all_definitions()`. Because it differs from what we have in smart contracts. IMO there is nothing wrong in constructing a query like this:

```suggestion
    let vec = client
        .query(FindAllAssetDefinitions::new())
```

Definitely not intended for this PR, but would like to get an opinion on this

Reply:
I have no emotional attachment to these helpers, but I think a similar syntax can be used if we decide to ever provide helpers for constructing pre-made queries with filters, to make actions like "get object by primary key" less boilerplaty. Although this API won't work as-is, not sure how exactly it would look like (and if we even want those boilerplate-reduction utilities).

Reply:
#4892